crates/vsock/src/vhu_vsock.rs

```diff
@@ -273,7 +273,7 @@ impl VhostUserBackendMut<VringRwLock, ()> for VhostUserVsockBackend {
             return Err(Error::HandleEventNotEpollIn.into());
         }
 
-        let mut thread = self.threads[thread_id].lock().unwrap();
+        let mut thread = self.threads[thread_id].write().unwrap();
```

Comment:
> Can you check if it is possible to take read lock here (changing process_tx, process_backend_evt, and process_rx)?

I don't think that's possible. Calls to each of `process_tx`, `process_backend_evt` and `process_rx` may require modifying the hashmaps contained in the thread's `thread_backend`.

Reply:
I see, it would be nice to check if we can use `RwLock` for them, but maybe is too big as refactoring and we don't  have much benefit. It would be nice though to have 2 separate threads to handle tx and rx that can run in parallel.

That said, if we can't take the read lock here, I think it's not worth replacing `Mutex`.